src/components/Announcements/Announcements.tsx

```diff
@@ -0,0 +1,184 @@
+import { useRouter } from 'next/router';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { getApolloContext } from '@apollo/client';
+import { DateTime } from 'luxon';
+import { DisplayMethodEnum } from 'pages/api/graphql-rest.page.generated';
+import { AlertBanner } from 'src/components/Shared/alertBanner/AlertBanner';
+import {
+  ActionEnum,
+  ContactFilterStatusEnum,
+} from 'src/graphql/types.generated';
+import { useAccountListId } from 'src/hooks/useAccountListId';
+import { useContactPartnershipStatuses } from 'src/hooks/useContactPartnershipStatuses';
+import { dispatch } from 'src/lib/analytics';
+import i18n from 'src/lib/i18n';
+import { DynamicAddAppealModal } from '../Tool/Appeal/Modals/AddAppealModal/DynamicAddAppealModal';
+import { DynamicAnnouncementBanner } from './AnnouncementBanner/DynamicAnnouncementBanner';
+import { DynamicAnnouncementModal } from './AnnouncementModal/DynamicAnnouncementModal';
+import {
+  ActionFragment,
+  useAcknowledgeAnnouncementMutation,
+  useAnnouncementsQuery,
+} from './Announcements.generated';
+
+const appealName = `${DateTime.local().year} ${i18n.t('End of Year Ask')}`;
```

Comment:
If we move this inside of the component, it will update if the announcement is shown on preferences and the user changes their language.